packages/use-query-params/src/__tests__/index.js

```diff
@@ -0,0 +1,149 @@
+    const { result } = renderHook(() => useQueryParam(), {
+      wrapper: MemoryRouter,
+    })
+
+    act(() => {
+      result.current.setQueryParams({
+        user: 'John Doe',
+        name: 'John',
+        lastName: 'Doe',
+        version: '1234',
+        lib: 'useQueryParams',
+      })
+    })
+    expect(result.current.queryParams).toEqual({
+      user: 'John Doe',
+      name: 'John',
+      lastName: 'Doe',
+      version: '1234',
+      lib: 'useQueryParams',
+    })
+  })
+
+  it('should get queryParams from existing location', () => {
+    const { result } = renderHook(() => useQueryParam(), {
+      // eslint-disable-next-line react/prop-types
+      wrapper: ({ children }) => (
+        <MemoryRouter
+          initialIndex={0}
+          initialEntries={[
+            { pathname: 'one', search: 'user=john' },
+            'two',
+            'three',
+          ]}
+        >
+          {children}
+        </MemoryRouter>
+      ),
+    })
+    expect(result.current.queryParams).toEqual({ user: 'john' })
+  })
+
+  it('should get queryParams from existing location and set new params', () => {
+    const { result } = renderHook(() => useQueryParam(), {
+      // eslint-disable-next-line react/prop-types
+      wrapper: ({ children }) => (
+        <MemoryRouter
+          initialIndex={0}
+          initialEntries={[
+            { pathname: 'one', search: 'user=john' },
+            'two',
+            'three',
+          ]}
+        >
+          {children}
+        </MemoryRouter>
+      ),
+    })
+    expect(result.current.queryParams).toEqual({ user: 'john' })
+    act(() => {
+      result.current.setQueryParams({
+        lastName: 'Doe',
+        version: '1234',
+        lib: 'useQueryParams',
+      })
+    })
+    expect(result.current.queryParams).toEqual({
+      user: 'john',
+      lastName: 'Doe',
+      version: '1234',
+      lib: 'useQueryParams',
+    })
+  })
+
+  it('should modify updater and erase old params', () => {
+    const { result } = renderHook(() => useQueryParam(), {
+      // eslint-disable-next-line react/prop-types
+      wrapper: ({ children }) => (
+        <MemoryRouter
+          initialIndex={0}
+          initialEntries={[
+            { pathname: 'one', search: 'user=john' },
+            'two',
+            'three',
+          ]}
+        >
+          {children}
+        </MemoryRouter>
+      ),
+    })
+    expect(result.current.queryParams).toEqual({ user: 'john' })
+    act(() => {
+      result.current.setQueryParams(() => ({
+        lastName: 'Doe',
+      }))
+    })
+    expect(result.current.queryParams).toEqual({
+      lastName: 'Doe',
+    })
+  })
+
+  it('should not update when location change', () => {
+    const { result } = renderHook(
+      () => useQueryParam({ updateOnChange: false }),
+      {
+        // eslint-disable-next-line react/prop-types
+        wrapper: ({ children }) => (
+          <MemoryRouter
```

Comment:
As this Wrapper is the same across multiple tests it would be nice to create a variable in top of the tests :)